src/orchestrator-tracing/src/lib.rs

```diff
@@ -129,6 +129,17 @@ pub struct TracingCliArgs {
         use_value_delimiter = true
     )]
     pub opentelemetry_resource: Vec<KeyValueArg<String, String>>,
+    /// Default the OpenTelemetry tracing collector to off, but allow it to be
+    /// dynamically turned on.
+    ///
+    /// Requires that the `--opentelemetry-endpoint` option is specified.
+    #[clap(
+        long,
+        env = "OPENTELEMETRTY_ENABLED",
+        default_value_t = DefaultTrue::default(),
+        requires = "opentelemetry-endpoint"
+    )]
+    pub opentelemetry_enabled: DefaultTrue,
```

Comment:
Does that mean you have to start the server either with an otel endpoint (we do in cloud, but others won't) or with `--opentelemetry-enabled=false`? Seems a little weird to me, if so

Reply:
No endpoint: the layer is `None` and entirely skipped
Endpoint: the layer is `Some(otel_layer.filter(dynamic_filter))`
`--opentelemetry-enabled` controls the initial state of `dynamic_filter`, in cloud we'd want `--opentelemetry-enabled=false`